client/jetpack-onboarding/main.jsx

```diff
@@ -35,7 +37,25 @@ class JetpackOnboardingMain extends React.PureComponent {
 		stepName: STEPS.SITE_TITLE,
 	};
 
-	// TODO: Add lifecycle methods to redirect if no siteId
+	componentDidMount() {
+		const { siteId, siteSlug } = this.props;
+
+		// If we are missing the Jetpack onboarding credentials,
+		// redirect back to wp-admin so we can obtain them again.
+		if ( ! siteId && siteSlug ) {
+			const siteDomain = siteSlug.replace( '::', '/' );
+			const url = addQueryArgs(
+				{
+					page: 'jetpack',
+					action: 'onboard',
+					calypso_env: config( 'env_id' ),
```

Comment:
Maybe only add this if it's not `production`?

Reply:
A little more specificity wouldn't hurt? Omitting that would rely on what the default values are, and by specifically passing it we disregard the default values, which sounds good to me.